### PR TITLE
[KIWI-1647]CIC | Contract Testing | Refactoring Clients

### PR DESCRIPTION
**Pull request by hemavana**
### What changed

Refactor Dynamo DB client, SQS client and KMS Client to initialise mock clients based on an environment variable
Evidence video of all endpoints working as expected attached here for reference.

Api tests are passing, evidence attached too.
1647_api_test_result

### Why did it change

CIC Contract testing

### Issue tracking

- [KIWI-1647](https://govukverify.atlassian.net/browse/KIWI-1647)

[KIWI-1647]: https://govukverify.atlassian.net/browse/KIWI-1647?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/AccessTokenHandler.ts [hunk 1]
```diff
@@ -28,28 +28,14 @@ export class AccessToken implements LambdaInterface {
...
-
-			default:
-				logger.error("Requested resource does not exist", {
-					resource: event.resource,
-					messageCode: MessageCodes.RESOURCE_NOT_FOUND,
-				});
-				throw new AppError("Requested resource does not exist" + { resource: event.resource }, HttpCodesEnum.NOT_FOUND);
-
-		}
+
```
Comment (Caitlin-cooling) [inline src/AccessTokenHandler.ts:31]: Please can you address the warnings in this file?

File: src/AccessTokenHandler.ts [hunk 2]
```diff
@@ -28,28 +28,14 @@ export class AccessToken implements LambdaInterface {
 		// clear PersistentLogAttributes set by any previous invocation, and add lambda context for this invocation
 		logger.setPersistentLogAttributes({});
 		logger.addContext(context);
-
-		switch (event.resource) {
```
Comment (Caitlin-cooling) [inline src/AccessTokenHandler.ts:32]: Thanks for removing this stuff!

File: src/AccessTokenHandler.ts [hunk 3]
```diff
@@ -28,28 +28,14 @@ export class AccessToken implements LambdaInterface {
...
-				});
-				throw new AppError("Requested resource does not exist" + { resource: event.resource }, HttpCodesEnum.NOT_FOUND);
-
-		}
+		
+		try {
+			logger.info("Received token request", { requestId: event.requestContext.requestId });
+			return await AccessTokenRequestProcessor.getInstance(logger, metrics).processRequest(event);
+		} catch (error: any) {
+			logger.error({ message: "An error has occurred. ", error });
```
Comment (Caitlin-cooling) [inline src/AccessTokenHandler.ts:36]: Should we have a messageCode for this?

File: src/utils/EnvironmentVariables.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { Logger } from "@aws-lambda-powertools/logger";
```
Comment (Caitlin-cooling) [inline src/utils/EnvironmentVariables.ts:1]: There is a test file for this that can be copied over

File: src/services/AuthorizationRequestProcessor.ts [hunk 5]
```diff
@@ -4,18 +4,19 @@ import { Metrics, MetricUnits } from "@aws-lambda-powertools/metrics";
...
 import { buildCoreEventFields } from "../utils/TxmaEvent";
+import { checkEnvironmentVariable } from "../utils/EnvironmentVariables";
+import { EnvironmentVariables } from "../utils/Constants";
 
-const SESSION_TABLE = process.env.SESSION_TABLE;
-const TXMA_QUEUE_URL = process.env.TXMA_QUEUE_URL;
-const ISSUER = process.env.ISSUER!;
+// const SESSION_TABLE = process.env.SESSION_TABLE;
+// const TXMA_QUEUE_URL = process.env.TXMA_QUEUE_URL;
+// const ISSUER = process.env.ISSUER!;
```
Comment (Caitlin-cooling) [inline src/services/AuthorizationRequestProcessor.ts:19]: Please remove

File: src/utils/Constants.ts [hunk 6]
```diff
@@ -48,7 +48,7 @@ export class Constants {
 
     static readonly AUTHORIZATION_CODE_INDEX_NAME = "authCode-updated-index";
 
-    static readonly TOKEN_EXPIRY_SECONDS = 3600;
+    static readonly TOKEN_EXPIRY_SECONDS = 3.154e+9;
```
Comment (Caitlin-cooling) [inline src/utils/Constants.ts:51]: What's this for?

File: src/utils/ValidationHelper.ts [hunk 7]
```diff
@@ -36,8 +36,7 @@ export class ValidationHelper {
 		if (headerValue === null || headerValue === undefined) {
 			throw new AppError("Missing header: Authorization header value is missing or invalid auth_scheme", HttpCodesEnum.UNAUTHORIZED);
 		}
-		const authHeader = event.headers.Authorization as string;
-		if (authHeader !== null && !authHeader.includes(Constants.BEARER)) {
+		if (!headerValue.includes(Constants.BEARER + " ")) {
```
Comment (Caitlin-cooling) [inline src/utils/ValidationHelper.ts:39]: This caused an API test to break in F2F - have we confirmed this it doesn't do the same in CIC?

Comment (hemavana) [reply src/utils/ValidationHelper.ts:39]: Yeah i have verified the same, there isnt any similar test for this that i could find. Also Zac has ran the tests on this branch and all looks good, attached the result to the ticket.

File: src/utils/Constants.ts (hunk 6 above)
Comment (hemavana) [reply src/utils/Constants.ts]: yeah....had added this for generating a long expiry token, forgot to revert it. i have reverted it now.

File: src/utils/EnvironmentVariables.ts (hunk 4 above)
Comment (hemavana) [reply src/utils/EnvironmentVariables.ts:1]: Done.

**Review by Caitlin-cooling — APPROVED** [review]
Nice work!

**Pull request merged**